x-pack/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_hosts.ts

```diff
@@ -167,16 +168,31 @@ export const indexEndpointHostDocs = usageTracker.track(
         if (!wasAgentEnrolled) {
           wasAgentEnrolled = true;
 
-          const { agents, fleetAgentsIndex, operations } = buildFleetAgentBulkCreateOperations({
-            endpoints: [hostMetadata],
-            agentPolicyId: realPolicies[appliedPolicyId].policy_ids[0],
-            kibanaVersion,
+          const agentOperations: BuildFleetAgentBulkCreateOperationsResponse = {
+            agents: [],
+            fleetAgentsIndex: '',
+            operations: [],
+          };
+
+          realPolicies[appliedPolicyId].policy_ids.forEach((policyId) => {
+            const { agents, fleetAgentsIndex, operations } = buildFleetAgentBulkCreateOperations({
```

Comment:
calling `buildFleetAgentBulkCreateOperations` for each agent policy id in `policy_ids`